metarParser-services/src/main/java/io/github/mivek/service/MetarService.java

```
@@ -45,6 +45,9 @@ public Metar retrieveFromAirport(final String icao) throws ParseException, IOExc
         HttpResponse<Stream<String>> response = HttpClient.newBuilder()
                 .build()
                 .send(request, HttpResponse.BodyHandlers.ofLines());
+
+        if (response.statusCode() != 200) return null;
```

I would throw an error instead of just returning null